### PR TITLE
Allow toggling comma/dot as decimal/thousands

### PR DESCRIPTION
**Pull request by Naveen3Singh**
This pull request addresses https://github.com/SimpleMobileTools/Simple-Calculator/issues/96

Users can now toggle between comma/dot as the default decimal separator

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calculator/activities/MainActivity.kt [hunk 1]
```diff
@@ -50,7 +53,9 @@ class MainActivity : SimpleActivity(), Calculator {
         btn_clear.setOnLongClickListener { calc.handleReset(); true }
 
         getButtonIds().forEach {
-            it.setOnClickListener { calc.numpadClicked(it.id); checkHaptic(it) }
+            it.setOnClickListener { view ->
+                calc.numpadClicked(view.id); checkHaptic(view)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/activities/MainActivity.kt:57]: you can aswell move this to 2 lines then, without needing ;

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/NumberFormatHelper.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        formatter.decimalFormatSymbols = symbols
+        formatter.isGroupingUsed = true
+        return formatter.format(d)
+    }
+
+    fun addGroupingSeparators(str: String) = doubleToString(
+        removeGroupingSeparator(str).toDouble()
+    )
+
+    fun removeGroupingSeparator(str: String) =
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/helpers/NumberFormatHelper.kt:28]: make this 1 line, looks weird now

**Comment by tibbi** [comment]
it doesnt work properly at the formula above the result, it is messed up and thousands separator is missing

**Comment by Naveen3Singh** [comment]
> it doesnt work properly at the formula above the result, it is messed up and thousands separator is missing

fixed in https://github.com/SimpleMobileTools/Simple-Calculator/pull/257/commits/2af2a6175ff40b1d39ed6e53b0bfc9ecb3b58da8

File: app/src/main/kotlin/com/simplemobiletools/calculator/extensions/Context.kt [hunk 3]
```diff
@@ -16,11 +19,19 @@ val Context.calculatorDB: CalculatorDao get() = CalculatorDatabase.getInstance(a
...
+        .forEach {
+            when (it) {
+                is TextView -> it.setTextColor(textColor)
+                is Button -> it.setTextColor(textColor)
+                is ViewGroup -> updateViewColors(it, textColor)
             }
+        }
+}
+
+inline fun <reified T> Context.refreshAppWidget() {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/extensions/Context.kt:31]: we have Context.updateWidgets() functions in some other apps including calendar, please use those, This solution looks strange.

Comment (Naveen3Singh) [reply app/src/main/kotlin/com/simplemobiletools/calculator/extensions/Context.kt]: done!

**Comment by tibbi** [comment]
should be good, thanks

**Pull request merged**